src/core/public/chrome/ui/header/nav_link.tsx

```diff
@@ -57,7 +59,8 @@ export function createEuiListItem({
         !isModifiedOrPrevented(event)
       ) {
         event.preventDefault();
-        navigateToApp(id);
+        if (deepLinkPath) navigateToUrl(deepLinkPath);
+        else navigateToApp(id);
```

Comment:
The deep links are not registered as applications so calling the `navigateToApp` with the deepLink.id didn't work here. 
I decided to add a new property I called `deepLinkPath` to the _ChromeNavLink_ in order to know the internal path of the link and call `navigateToUrl` instead, which will call `navigateToApp` internally with the proper app id and path.

Reply:
Should be fine to use `navigateToUrl`. I was afraid it may break the`defaultPath` of the application:

https://github.com/elastic/kibana/blob/206a64fc4cb96d66bf1a32e683817deeeea10598/src/core/public/application/application_service.tsx#L238-L240

https://github.com/elastic/kibana/blob/206a64fc4cb96d66bf1a32e683817deeeea10598/src/core/public/application/application_service.tsx#L281-L286

However `parseAppUrl` properly returns `undefined` if the path is empty, so it seems ok.

https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/core/public/application/utils/parse_app_url.ts#L68